src/batch/src/executor/s3_file_scan.rs

```diff
@@ -113,3 +116,35 @@ impl S3FileScanExecutor {
         }
     }
 }
+
+pub struct FileScanExecutorBuilder {}
+
+#[async_trait::async_trait]
+impl BoxedExecutorBuilder for FileScanExecutorBuilder {
+    async fn new_boxed_executor<C: BatchTaskContext>(
+        source: &ExecutorBuilder<'_, C>,
+        _inputs: Vec<BoxedExecutor>,
+    ) -> crate::error::Result<BoxedExecutor> {
+        let file_scan_node = try_match_expand!(
+            source.plan_node().get_node_body().unwrap(),
+            NodeBody::FileScan
+        )?;
+
+        assert_eq!(file_scan_node.storage_type, StorageType::S3 as i32);
+
+        Ok(Box::new(S3FileScanExecutor::new(
+            #[allow(deprecated)]
```

Comment:
What is `deprecated` here?

Reply:
Oh, I used a deprecated method `from_i32`. Let me change it to `try_from`